### PR TITLE
test: add tests and fix accessibility issues for the select

### LINKED ISSUE #263: Accessibility documentation for select component
**Is your feature request related to a problem? Please describe.**
Like all components, the documentation of the select should contain how we make sure the widget is accessible and what are the aria fields that can be set.

### LINKED ISSUE #459: select: add accessibility tests

### PR DESCRIPTION
**Pull request by fbasso**
fix #459
fix #263

### CODE REVIEW

File: angular/lib/src/components/select/select.component.ts [hunk 1]
```diff
@@ -75,14 +76,16 @@ export class SelectItemDirective<Item> {
...
 					[auUse]="menuDirective"
 					[class]="'dropdown-menu show ' + (menuClassName || '')"
 					[attr.data-popper-placement]="state.placement"
 					(mousedown)="$event.preventDefault()"
 				>
 					@for (itemContext of state.visibleItems; track itemCtxTrackBy($index, itemContext)) {
 						<li
-							[class]="'au-select-item dropdown-item position-relative' + (itemContext === state.highlighted ? ' bg-primary text-light' : '')"
+							[class]="'au-select-item dropdown-item position-relative'"
+							[class.bg-text-bg-primary]="itemContext === state.highlighted"
```
Comment (ExFlo) [inline angular/lib/src/components/select/select.component.ts:88]: is this this good in dark mode ?

Comment (fbasso) [reply angular/lib/src/components/select/select.component.ts:88]: I don't know what's happened, but the point is that the bg-text-bg-primary class is not included anymore :(

And it should be fine for the dark mode, as it's like this Bootstrap manages it (there is no color switch in bootstrap. For example https://getbootstrap.com/docs/5.3/components/toasts/#color-schemes).

@ExFlo, how is it supposed to be done ?

Comment (fbasso) [reply angular/lib/src/components/select/select.component.ts:88]: Oh, my bad, it's `text-bg-primary` that needs to be used !

File: angular/lib/src/components/select/select.component.ts [hunk 2]
```diff
@@ -75,14 +76,19 @@ export class SelectItemDirective<Item> {
...
 					[class]="'dropdown-menu show ' + (menuClassName || '')"
 					[attr.data-popper-placement]="state.placement"
 					(mousedown)="$event.preventDefault()"
 				>
 					@for (itemContext of state.visibleItems; track itemCtxTrackBy($index, itemContext)) {
 						<li
-							[class]="'au-select-item dropdown-item position-relative' + (itemContext === state.highlighted ? ' bg-primary text-light' : '')"
+							role="option"
+							[attr.aria-selected]="itemContext.selected"
+							[class]="'au-select-item dropdown-item position-relative'"
```
Comment (quentinderoubaix) [inline angular/lib/src/components/select/select.component.ts:90]: Could be just
```class="au-select-item dropdown-item position-relative"```

File: core/src/utils/internal/dom.ts [hunk 3]
```diff
@@ -87,3 +87,11 @@ export function addEvent(element: Element, type: string, fn: EventListenerOrEven
...
 	};
 }
+
+let idCount = 0;
+/**
+ * Generates a unique ID with the format 'auId-[counter]'.
+ *
+ * @returns The generated ID.
+ */
+export const getGeneratedId = () => `auId-${idCount++}`;
```
Comment (quentinderoubaix) [inline core/src/utils/internal/dom.ts:97]: maybe rename to **generateId** ?
I usually prefer to add the *get* prefix only when the rest of the name does not clarify that it returns something.

another thing: after merging this, create an issue to use this in the accordion item, which also manages a unique id.

File: e2e/select/select.chromium.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import {analyze} from '../utils';
+
+test.describe('Accessibility', () => {
+	test('Opened select', async ({page, browserName}) => {
+		if (browserName === 'chromium') {
+			await page.goto('#/select/default');
+			const selectPO = new SelectPO(page);
+			const locatorInput = selectPO.locatorInput;
+			await locatorInput.fill('a');
+			await locatorInput.press('Enter');
```
Comment (quentinderoubaix) [inline e2e/select/select.chromium.e2e-spec.ts:12]: maybe add an expect before the axe tests to make sure the popup is opened.

File: e2e/select/select.chromium.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+			expect((await analyze(page, page.url())).violations).toEqual([]);
```
Comment (quentinderoubaix) [inline e2e/select/select.chromium.e2e-spec.ts:13]: Note that in the main accessibility test, we test for both light and dark modes.
I wonder if we should do the same here...

File: svelte/lib/src/components/select/Select.svelte [hunk 6]
```diff
@@ -55,15 +55,17 @@
 		directives: {floatingDirective, hasFocusDirective, referenceDirective, inputContainerDirective},
 	} = widget;
 	$: widget.patchChangedProps($$props);
+
+	$: menuId = `${$id$}-menu`;
```
Comment (quentinderoubaix) [inline svelte/lib/src/components/select/Select.svelte:59]: a lot of dollars 😄

**Review by quentinderoubaix — COMMENTED** [review]
@fbasso left some minor comments, thx for the PR !

File: angular/lib/src/components/select/select.component.ts (hunk 2 above)
Comment (fbasso) [reply angular/lib/src/components/select/select.component.ts:90]: Right :)

File: e2e/select/select.chromium.e2e-spec.ts (hunk 4 above)
Comment (fbasso) [reply e2e/select/select.chromium.e2e-spec.ts:12]: Done ;). I've added more by the way:

- the first one to be sure the first item is selected,
- the second one to be sure that the dropdown is still open.

**Comment by fbasso** [comment]
@quentinderoubaix, pushed  again with changes.

File: e2e/utils.ts [hunk 7]
```diff
@@ -1,4 +1,30 @@
...
+ * @param page
+ * @param route
+ * @returns
+ */
+export async function analyze(page: Page, route: string): Promise<AxeResults> {
+	const analyser = new AxeBuilder({page}).withTags(['wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa']);
+	return analyser.analyze();
+}
+
+export async function swithTheme(page: Page, theme: 'light' | 'dark') {
```
Comment (quentinderoubaix) [inline e2e/utils.ts:21]: little typo, should be ```switchTheme```

**Review by quentinderoubaix — APPROVED** [review]
thanks for the changes @fbasso !

File: e2e/utils.ts (hunk 7 above)
Comment (fbasso) [reply e2e/utils.ts:21]: Thanks !

**Comment by fbasso** [comment]
Thanks @quentinderoubaix

**Pull request merged**